src/core/public/analytics/analytics_service.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { AnalyticsClient } from '@elastic/analytics';
+import { createAnalytics } from '@elastic/analytics';
+import { CoreContext } from '../core_system';
+import { createLogger } from './logger';
+
+/**
+ * Exposes the public APIs of the AnalyticsClient during the setup phase.
+ * {@link AnalyticsClient}
+ * @public
+ */
+export type AnalyticsServiceSetup = AnalyticsClient;
+/**
+ * Exposes the public APIs of the AnalyticsClient during the start phase
+ * {@link AnalyticsClient}
+ * @public
+ */
+export type AnalyticsServiceStart = Pick<
+  AnalyticsClient,
+  'optIn' | 'reportEvent' | 'telemetryCounter$'
+>;
+
+export class AnalyticsService {
+  private readonly analyticsClient: AnalyticsClient;
+
+  constructor(core: CoreContext) {
+    this.analyticsClient = createAnalytics({
+      isDev: core.env.mode.dev,
+      logger: createLogger(core.env.mode.dev),
+      // TODO: We need to be able to edit sendTo once we resolve the telemetry config.
+      //  For now, we are relying on whether it's a distributable or running from source.
+      sendTo: core.env.packageInfo.dist ? 'production' : 'staging',
```

Comment:
Can definitely be done in a follow-up, but thinking out loud, I wonder how we will be able to access the telemetry config from `core`.  Should we expose an API for another actor to call it with the config? Should core access the telemetry config directly (it's hacky and a bad isolation of concern, but technically core can access any plugin's config...)

Reply:
Yeah! I'm still wrapping my head about the best way to spread the config given the architecture of Kibana.

I've created #129014 to discuss and find out our preferred approach.